src/common/NativeObject.ts

```diff
@@ -0,0 +1,4 @@
+/**
+ * Represents native objects (Date)
+ */
+export type NativeObject = Date
```

Comment:
this extra type is definitely redundant

Reply:
the idea was to have a type that is going to represent a set of native types. Initially, I set it to `Date | Buffer`:

I.e. it would be 
```
          : T extends NativeObject
          ? T
```
instead of 
```
          : T extends Date
          ? T
          : T extends Buffer
          ? T
```
but then I removed Buffer because it's not native to browsers. 
Suggest keeping NativeObject type in case we add more in the future 😄

Reply:
I understand, but for the simplification purposes we don't need another type.